Sources/Architecture/ARM/SoftwareBreakpointManager.cpp

```diff
@@ -35,7 +35,10 @@ void SoftwareBreakpointManager::clear() {
 }
 
 ErrorCode SoftwareBreakpointManager::add(Address const &address, Type type,
-                                         size_t size) {
+                                         size_t size, Mode mode) {
+  if (mode != kModeExec)
```

Comment:
Ah true.

On Friday, April 8, 2016, Francis Ricci <notifications@github.com> wrote:

> In Sources/Architecture/ARM/SoftwareBreakpointManager.cpp
> <https://github.com/facebook/ds2/pull/243#discussion_r59102598>:
>
> > @@ -35,7 +35,10 @@ void SoftwareBreakpointManager::clear() {
> >  }
> >
> >  ErrorCode SoftwareBreakpointManager::add(Address const &address, Type type,
> > -                                         size_t size) {
> > +                                         size_t size, Mode mode) {
> > +  if (mode != kModeExec)
>
> There is no SoftwareBreakpointManager superclass, "super" here is
> BreakpointManager, and I can't put it there, or it will break hardware
> breakpoints. But I can change it to an assert.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/ds2/pull/243/files/2b7add545bce75bd539f7f39cb938aa59689d8ae#r59102598>
>


-- 
Stephane Sezer